saleor/graphql/api.py

```diff
@@ -101,35 +102,30 @@ class Meta:
         model = Category
         interfaces = (relay.Node, DjangoPkInterface)
 
-    def resolve_ancestors(self, args, context, info):
-        return get_ancestors_from_cache(self, context)
+    def resolve_ancestors(self, info):
+        return get_ancestors_from_cache(self, info.context)
 
-    def resolve_children(self, args, context, info):
+    def resolve_children(self, info):
         return self.children.all()
 
-    def resolve_siblings(self, args, context, info):
+    def resolve_siblings(self, info):
         return self.get_siblings()
 
-    def resolve_products_count(self, args, context, info):
+    def resolve_products_count(self, info):
         return self.products.count()
 
-    def resolve_url(self, args, context, info):
-        ancestors = get_ancestors_from_cache(self, context)
+    def resolve_url(self, info):
+        ancestors = get_ancestors_from_cache(self, info.context)
         return self.get_absolute_url(ancestors)
 
-    def resolve_products(self, args, context, info):
-
-        def filter_by_price(queryset, value, operator):
-            return [obj for obj in queryset if operator(get_availability(
-                obj, context.discounts).price_range.min_price.gross, value)]
-
-        tree = self.get_descendants(include_self=True)
+    def resolve_products(self, info, **args):
+        context = info.context
         qs = products_for_api(context.user)
-        qs = qs.filter(categories__in=tree).distinct()
-        attributes_filter = args.get('attributes')
-        order_by = args.get('order_by')
-        price_lte = args.get('price_lte')
-        price_gte = args.get('price_gte')
+        tree = self.get_descendants(include_self=True)
+        qs = qs.filter(categories__in=tree)
```

Comment:
That `distinct()` was here to return proper results when there are multiple categories of the same name (this was added in PR #1331).